src/passport-saml/saml.ts

```diff
@@ -730,6 +730,8 @@ class SAML {
     if (totalReferencedNodes.length > 1) {
       return false;
     }
+    // normalize XML to replace XML-encoded carriage returns with actual carriage returns
+    fullXml = this.normalizeXml(fullXml);
     fullXml = this.normalizeNewlines(fullXml);
```

Comment:
Now that we are using `xmldom` to normalize, do we need to call `normalizeNewlines()`?

Reply:
Yes, we still need it. If we remove `normalizeNewlines()`, the tests will fail. `xmldom` will normalize the encoded carriage returns into `\r`, but we (as the XML processor) are still responsible for normalizing the `\r` into `\n`.